kotlin/navigate-route-with-rerouting/src/main/java/com/esri/arcgisruntime/sample/navigateroutewithrerouting/MainActivity.kt

```diff
@@ -0,0 +1,396 @@
+                    0.0
+                )
+                // set the locations to travel using the JSON polylines
+                simulatedLocationDataSource.setLocations(
+                    Geometry.fromJson(polylineJSON) as Polyline,
+                    simulationParameters
+                )
+                // provide the route tacker with the current location of the simulated datasource
+                simulatedLocationDataSource.addLocationChangedListener {
+                    routeTracker?.trackLocationAsync(it.location)
+                }
+                // add a data source for the location display
+                mapView.locationDisplay.locationDataSource = simulatedLocationDataSource
+                // use this instead if you want real location:
+                //mapView.locationDisplay.locationDataSource = RouteTrackerLocationDataSource(routeTracker, AndroidLocationDataSource(this))
+
+                // start the location data source
+                mapView.locationDisplay.startAsync()
+                // add re-routing listeners
+                routeTracker?.addRerouteStartedListener(routeStartedListener)
+                routeTracker?.addRerouteCompletedListener(routeCompletedListener)
+            }
+        }
+    }
+
+    /**
+     * Removes the current route tracking status listener if new route loading has started
+     */
+    private val routeStartedListener: RouteTracker.RerouteStartedListener =
+        RouteTracker.RerouteStartedListener {
+            routeTracker?.removeTrackingStatusChangedListener(trackingStatusListener)
+        }
+
+    /**
+     * Removes the route tracking status listener if re-route loading has completed,
+     * sets up new direction maneuvers and adds a new route tracking listener
+     */
+    private val routeCompletedListener: RouteTracker.RerouteCompletedListener =
+        RouteTracker.RerouteCompletedListener {
+            routeTracker?.removeTrackingStatusChangedListener(trackingStatusListener)
+            // clear current direction maneuvers
+            directionManeuvers.clear()
+            // adds new direction maneuvers to the list
+            directionManeuvers.addAll(it.trackingStatus.routeResult.routes[0].directionManeuvers)
+            // re-add the event listeners for tracking status changes
+            routeTracker?.addTrackingStatusChangedListener(trackingStatusListener)
+        }
+
+
+    /**
+     * Calls updateTrackingStatus() on every route tacking status changed
+     */
+    private val trackingStatusListener: RouteTracker.TrackingStatusChangedListener =
+        RouteTracker.TrackingStatusChangedListener {
+            updateTrackingStatus(it.trackingStatus)
+        }
+
+    /**
+     * Displays the route status descriptions and update route graphics as the
+     * the datasource updates on the map
+     */
+    private fun updateTrackingStatus(status: TrackingStatus) {
+        val statusMessage: StringBuilder = StringBuilder("")
```

Comment:
```suggestion
        val statusMessage: StringBuilder = StringBuilder()
```

StringBuilder defaults to an empty string anyways I think